### DISCUSSION TITLE
VRP distance matrix

### DISCUSSION THREAD

**Discussion by rahulj2409** [google/or-tools#2642] [Routing (and legacy CP) questions]
It seems to be the case that when you do routing.GetArcCostForVehicle(prev, index, vehicle_id), the value returned is always an integer, regardless if the value itself in the distance matrix is a decimal/float value. Are there any methods that can handle this, as I am currently just calculating directly from the distance matrix itself to account for this but was wondering if there's a safer/less hack-y way of handling this. 

Thanks!

**Comment by rahulj2409** [comment]
Furthermore, I just want to make sure that the solution that is found by google OR tools is not rounding down the values in the matrix like this. If so, how can one make it use float values instead. Thanks!

**Comment by Mizux** [comment]
Solver only support integer value (e.d. `int64_t`). You can offset your value by 10, 100, 1000... to simulate fixed width (and symply divide by the same offset when parsing the solution)
